src/Chunk.ts

```diff
@@ -1011,7 +1011,13 @@ export default class Chunk {
 	}
 
 	generateIdPreserveModules(preserveModulesRelativeDir: string) {
-		return (this.id = normalize(relative(preserveModulesRelativeDir, this.entryModule.id)));
+		const sanitizedId = this.entryModule.id.replace('\0', '_');
+		this.id = normalize(
+			isAbsolute(this.entryModule.id)
+				? relative(preserveModulesRelativeDir, sanitizedId)
+				: 'ROLLUP_' + basename(sanitizedId)
```

Comment:
Perhaps make these a folder - `_virtual/...`, or something like that, in case there are a lot of them?